addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiActorGatewayHandler.java

```diff
@@ -0,0 +1,222 @@
+ */
+package org.openhab.binding.mihome.handler;
+
+import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.*;
+
+import java.util.Iterator;
+
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.mihome.internal.ColorUtil;
+
+import com.google.gson.JsonObject;
+
+/**
+ * @author Patrick Boos - Initial contribution
+ * @author Dimalo
+ */
+public class XiaomiActorGatewayHandler extends XiaomiActorBaseHandler {
+
+    private float lastBrightness = -1;
+
+    public XiaomiActorGatewayHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    void execute(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case CHANNEL_BRIGHTNESS:
+                if (command instanceof PercentType) {
+                    float newBright = ((PercentType) command).floatValue();
+                    if (lastBrightness != newBright) {
+                        lastBrightness = newBright;
+                        logger.debug("actual brigthness {}", lastBrightness);
+                        writeBridgeLightColor(getGatewayLightColor(), lastBrightness / 100);
+                    } else {
+                        logger.debug("Do not send this command, value {} already set", newBright);
+                    }
+                } else if (command instanceof OnOffType) {
+                    if (lastBrightness == -1) {
+                        try {
+                            Iterator<Item> iter = linkRegistry
+                                    .getLinkedItems(new ChannelUID(this.thing.getUID(), CHANNEL_BRIGHTNESS)).iterator();
+                            while (iter.hasNext()) {
+                                Item item = iter.next();
+                                if (item.getState() instanceof PercentType) {
+                                    lastBrightness = Float.parseFloat(item.getState().toString());
+                                    logger.debug("last brightness value found: {}", lastBrightness);
+                                    break;
+                                }
+                            }
+                            lastBrightness = lastBrightness == -1 || lastBrightness == 0 ? 1 : lastBrightness;
+                            logger.debug("No dimmer value for brightness found, adjusted to {}", lastBrightness);
+                        } catch (NumberFormatException e) {
+                            lastBrightness = 1;
+                            logger.debug("No last brightness value found - assuming 100");
```

Comment:
Okay, but on line 67 you log it as a value between -1 & 1, so it's at least inconsistent, maybe just log "assuming full brightness"